mailbox/store/src/test/java/org/apache/james/mailbox/store/mail/model/MailboxMapperTest.java

```diff
@@ -82,25 +82,60 @@ void setUp() {
     }
 
     @Test
-    void findMailboxByPathWhenAbsentShouldFail() throws MailboxException {
+    void findMailboxByPathWhenAbsentShouldFail() {
         assertThatThrownBy(() -> mailboxMapper.findMailboxByPath(MailboxPath.forUser(BENWA, "INBOX")))
             .isInstanceOf(MailboxNotFoundException.class);
     }
 
     @Test
-    void saveShouldPersistTheMailbox() throws MailboxException {
-        mailboxMapper.save(benwaInboxMailbox);
+    void createShouldPersistTheMailbox() throws MailboxException {
+        benwaInboxMailbox.setMailboxId(null);
+        mailboxMapper.create(benwaInboxMailbox);
+
+        MailboxAssert.assertThat(mailboxMapper.findMailboxByPath(benwaInboxPath)).isEqualTo(benwaInboxMailbox);
+        MailboxAssert.assertThat(mailboxMapper.findMailboxById(benwaInboxMailbox.getMailboxId())).isEqualTo(benwaInboxMailbox);
+    }
+
+    @Test
+    void createShouldThrowWhenMailboxAlreadyExists() throws MailboxException {
+        benwaInboxMailbox.setMailboxId(null);
+        mailboxMapper.create(benwaInboxMailbox);
+
+        Mailbox mailbox = new Mailbox(benwaInboxMailbox);
+        mailbox.setMailboxId(null);
+
+        assertThatThrownBy(() -> mailboxMapper.create(mailbox))
+            .isInstanceOf(MailboxExistsException.class);
+    }
+
+    @Test
+    void createShouldSetAMailboxIdForMailbox() throws MailboxException {
+        benwaInboxMailbox.setMailboxId(null);
+        MailboxId mailboxId = mailboxMapper.create(benwaInboxMailbox);
+
+        assertThat(mailboxId).isNotNull();
+    }
+
+    @Test
+    void createShouldThrowWhenMailboxIdNotNull() {
+        assertThatThrownBy(() -> mailboxMapper.create(benwaInboxMailbox))
+            .isInstanceOf(IllegalArgumentException.class);
+    }
+
+    @Test
+    void renameShouldPersistTheMailbox() throws MailboxException {
+        mailboxMapper.rename(benwaInboxMailbox);
         MailboxAssert.assertThat(mailboxMapper.findMailboxByPath(benwaInboxPath)).isEqualTo(benwaInboxMailbox);
     }
 
     @Test
-    void saveShouldThrowWhenMailboxAlreadyExist() throws MailboxException {
-        mailboxMapper.save(benwaInboxMailbox);
+    void renameShouldThrowWhenMailboxAlreadyExist() throws MailboxException {
```

Comment:
Although the implementation is doing it, but I feel like this sentence is weird. Normally, rename should always success when mailbox exists. 

My suggestion is rename this one into `renameShouldWorkWhenMailboxAlreadyExists` and disable it. And one more test I would like to have is `renameShouldThrowWhenMailboxDoesntExist`

Reply:
Mailbox rename should always succeed if the mailbox exists, are you sure?

* I have mailboxA and mailboxB
* I rename mailboxA to mailboxB... that should be allowed, really?

Reply:
Name is confusing but I update those tests in #3108